src/vulkan/engine_vulkan.cc

```diff
@@ -51,6 +51,40 @@ VkShaderStageFlagBits ToVkShaderStage(ShaderType type) {
   return VK_SHADER_STAGE_FRAGMENT_BIT;
 }
 
+bool IsFormatSupportedByPhysicalDevice(BufferType type,
+                                       VkPhysicalDevice physical_device,
+                                       VkFormat format) {
+  VkFormatProperties properties = {};
+  vkGetPhysicalDeviceFormatProperties(physical_device, format, &properties);
+
+  VkFormatFeatureFlagBits flags = VK_FORMAT_FEATURE_VERTEX_BUFFER_BIT;
+  switch (type) {
+    case BufferType::kColor:
+      flags = VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT;
+      break;
+    case BufferType::kDepth:
+      flags = VK_FORMAT_FEATURE_DEPTH_STENCIL_ATTACHMENT_BIT;
+      break;
+    case BufferType::kSampled:
+      flags = VK_FORMAT_FEATURE_SAMPLED_IMAGE_BIT;
+      break;
+    case BufferType::kVertex:
+      flags = VK_FORMAT_FEATURE_VERTEX_BUFFER_BIT;
+      break;
+    default:
+      return false;
+  }
+
+  return (properties.bufferFeatures & flags) == flags;
+}
+
+bool ExceedMaximumDescriptorSets(VkPhysicalDevice physical_device,
```

Comment:
Let's invert this to:

IsDescriptorSetInBounds

Reply:
done